book/build.py

```diff
@@ -0,0 +1,54 @@
+import subprocess
+from pathlib import Path
+import urllib.parse
+
+
+SCRIPT_DIR = Path(__file__).parent
+
+
+def generate_example(filename, title, strip_assets):
+    path_in = SCRIPT_DIR.parent / "examples" / f"{filename}.nbt"
+    path_out = SCRIPT_DIR / "src" / f"example-{filename}.md"
+
+    code = []
+    with open(path_in, "r") as fin:
+        for line in fin:
+            if not strip_assets or "assert_eq" not in line:
+                code.append(line)
+
+    url = f"https://numbat.dev/?q={urllib.parse.quote_plus(''.join(code))}"
+
+    with open(path_out, "w") as fout:
+        fout.write("<!-- This file is autogenerated! Do not modify it -->\n")
+        fout.write("\n")
+        fout.write(f'# {title} <a href="{url}"><i class="fa fa-play"></i></a>\n')
+        fout.write("\n")
+        fout.write("``` numbat\n")
+        fout.writelines(code)
+        fout.write("```\n")
+
+
+generate_example("acidity", "Acidity", True)
+generate_example("barometric_formula", "Barometric formula", True)
+generate_example("body_mass_index", "Body mass index", True)
+generate_example("factorial", "Factorial", False)
+generate_example("medication_dosage", "Medication dosage", True)
+generate_example("molarity", "Molarity", True)
+generate_example("musical_note_frequency", "Musical note frequency", True)
+generate_example("paper_size", "Paper sizes", False)
+generate_example("pipe_flow_rate", "Flow rate in a pipe", True)
+generate_example("population_growth", "Population growth", True)
+generate_example("recipe", "Recipe", True)
+generate_example("voyager", "Voyager", True)
+generate_example("xkcd_687", "XKCD 687", True)
+generate_example("xkcd_2585", "XKCD 2585", True)
+generate_example("xkcd_2812", "XKCD 2812", True)
+generate_example("workhours", "Workhours", True)
+
+generate_example("numbat_syntax", "Syntax overview", False)
```

Comment:
Running this syntax overview (all at once) does not make a lot of sense. Can we skip the "Run" link for this page? We could add a fourth (defaulted) argument to `generate_example`

Reply:
Agreed, this is a bit much of a wall of code. I added a parameter to turn the link insertion off.